client/src/pages/ChatRoom.jsx

```diff
@@ -1,17 +1,37 @@
-import React from "react";
-import ChatBox from "../components/ChatBox";
-import SendMessage from "../components/SendMessage";
-import { MessagesProvider } from "../MessagesContext";
+import React, { useState, useEffect } from 'react';
+import ChatBox from '../components/ChatBox';
+import SendMessage from '../components/SendMessage';
+import { MessagesProvider } from '../MessagesContext';
+import { useNavigate } from 'react-router-dom';
 
 const ChatRoom = () => {
+  const [isSet, setIsSet] = useState(false);
```

Comment:
use more meaningful names to make it easier to understand the intention behind the code @blacky-sh "isSet", "setIsSet" can be renamed "hasAllSettingsDefined" and "setHasAllSettingsDefined"

Reply:
Noted! we will refactor it asap.